src/cli/path_validation.py

```
@@ -0,0 +1,127 @@
+        ``.is_absolute()`` and a fully-normalised form.
+
+    Raises:
+        typer.BadParameter: Missing path (when ``must_exist=True``) or
+            path-exists-but-not-a-directory (when ``must_be_dir=True``).
+            Typer renders these as ``Usage: ... Invalid value ...``
+            rather than a Python traceback.
+    """
+    resolved = path.expanduser().resolve()
+
+    if must_exist and not resolved.exists():
+        raise typer.BadParameter(f"Path does not exist: {path!s} (resolved to {resolved!s})")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)